packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/ELKDiagramConverter.java

```diff
@@ -100,24 +105,33 @@ private ElkNode convertDiagram(Diagram diagram) {
         return elkDiagram;
     }
 
-    private void convertNode(Node node, ElkNode parent, Map<String, ElkConnectableShape> connectableShapeIndex, Map<String, ElkGraphElement> id2ElkGraphElements) {
+    private void convertNode(Node node, ElkNode parent, Map<String, ElkConnectableShape> connectableShapeIndex, Map<String, ElkGraphElement> id2ElkGraphElements,
+            ISiriusWebLayoutConfigurator layoutConfigurator) {
         ElkNode elkNode = ElkGraphFactory.eINSTANCE.createElkNode();
         elkNode.setIdentifier(node.getId());
         elkNode.setProperty(PROPERTY_TYPE, node.getType());
         if (node.getChildrenLayoutStrategy() != null) {
             elkNode.setProperty(PROPERTY_CHILDREN_LAYOUT_STRATEGY, node.getChildrenLayoutStrategy().getClass());
         }
 
-        TextBounds textBounds = this.textBoundsService.getBounds(node.getLabel());
+        TextBounds textBounds = null;
+        Label label = node.getLabel();
+        String labelType = this.getNodeLabelType(node, layoutConfigurator);
```

Comment:
It seems `labelType` is only used to test if it `startsWith("label:inside-v")` just below, and never after after this. Do we need the whole `getNodeLabelType` and associated methods for that info?

Reply:
Yes I think we do.